src/components/label/Label.tsx

```diff
@@ -7,38 +7,38 @@ import cn from 'classnames';
 
 import classes from 'src/components/label/Label.module.css';
 import { LabelContent } from 'src/components/label/LabelContent';
+import { useFormComponentCtx } from 'src/layout/FormComponentContext';
 import { gridBreakpoints } from 'src/utils/formComponentUtils';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
 import type { LabelContentProps } from 'src/components/label/LabelContent';
-import type { IGridStyling, ILabelSettings } from 'src/layout/common.generated';
+import type { ExprResolved } from 'src/features/expressions/types';
+import type { IGridStyling, TRBLabel } from 'src/layout/common.generated';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
 
 type LabelType = 'legend' | 'span' | 'label';
 
 export type LabelProps = PropsWithChildren<{
-  id: string;
+  node: LayoutNode;
   renderLabelAs: LabelType;
-  required?: boolean;
-  readOnly?: boolean;
-  labelSettings?: ILabelSettings;
-  grid?: { labelGrid?: IGridStyling };
-  textResourceBindings?: {
-    title?: string;
-    description?: string;
-    help?: string;
-  };
   className?: string;
+
+  id?: string;
+  textResourceBindings?: ExprResolved<TRBLabel>;
 }>;
 
-export function Label({
-  id,
-  renderLabelAs,
-  required,
-  readOnly,
-  labelSettings,
-  grid,
-  textResourceBindings,
-  children,
-  className,
-}: LabelProps) {
+export function Label({ node, renderLabelAs, children, className, ...rest }: LabelProps) {
+  const overrideItemProps = useFormComponentCtx()?.overrideItemProps;
+  const _item = useNodeItem(node);
+  const item = { ..._item, ...overrideItemProps };
+  const { id: nodeId, grid, textResourceBindings: _trb } = item;
+  const required = 'required' in item && item.required;
+  const readOnly = 'readOnly' in item && item.readOnly;
+  const labelSettings = 'labelSettings' in item ? item.labelSettings : undefined;
+
+  // These can be overridden by props, but are otherwise retrieved from the node item
+  const id = rest.id ?? nodeId;
+  const textResourceBindings = (rest.textResourceBindings ?? _trb) as ExprResolved<TRBLabel> | undefined;
```

Comment:
TS casting resulterer fort i bugs fordi `rest.textResourceBindings` eller `_trb` plutselig kan være noe helt annet enn `ExprResolved<TRBLabel> | undefined`, og da finner man ikke ut av det før runtime :/ 
Så `as` burde egentlig aldri brukes, med mindre man vet 100% hvilken type noe er (og har compile time checks), men at TS bare ikke har klart å skjønne hvilken type det er.
Supert om du får til å fjerne dem (nye altså), så blir det også lettere å refaktorere senere 😃

Reply:
Ja, jeg tenkte litt på det, men jeg gjorde ikke så mye med det (også for å ikke bare introdusere enda mer greier her i denne PRen). Slik det var før fikk man heller ikke noe feil om man sendte inn `textResourceBindings` til en komponent-type som ikke har dem definert (siden `LabelProps` gjorde `title`, `description` og `help` valgfri, men en komponent helt uten de tekstressursbindingene ville da også tilsynelatende fungert.

Jeg ser for meg at jeg heller ville laget noe ala `CompWithBehavior` her, og sagt at en label bare kan få inn node-objekter for komponent-typer som har blitt konfigurert til å tillate `TRBLabel`. Først da får man en TS-feil om man har glemt å konfigurere opp label-tekstressurser på en komponent man vil rendre label for.. :thinking:

Jeg ordner et issue på dette istedenfor! :raised_hands:

Reply:
https://github.com/Altinn/app-frontend-react/issues/2315

Reply:
Denne kommentaren var ment for generelt bruk av `as` som er gjort mange andre steder også. Supert om du får fikset det 😊